source/neuropod/tests/test_ope_multiple_instances.cc

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "neuropod/neuropod.hh"
+#include "neuropod/tests/ope_overrides.hh"
+
+#include <thread>
+
+TEST(test_ope_multiple_instances, multithreaded)
+{
+    const auto NUM_INSTANCES = 4;
+    const auto NUM_REQUESTS  = 2048;
```

Comment:
I'd declare both as size_t explicitly (size_t is a right type for num/size/index in 64bit world) and change then iterator to size_t in a for loop below:
```
for (unsigned long i = 0; i < NUM_INSTANCES; ++i)

Reply:
This is heavily based off of the repro code you provided in #397. I missed the `unsigned long` when I was turning it into a test. Will update.

Don't think the explicit `size_t` is necessary.